src/SessionSetup/include/SessionSetup/RetrieveInstancesWidget.h

```diff
@@ -54,7 +56,7 @@ class RetrieveInstancesWidget : public QWidget {
 
   std::unique_ptr<Ui::RetrieveInstancesWidget> ui_;
   std::shared_ptr<MainThreadExecutor> main_thread_executor_;
-  RetrieveInstances* retrieve_instances_;
+  std::shared_ptr<RetrieveInstances> retrieve_instances_;
```

Comment:
I am not completely convinced that this should be a shared pointer. Either `main` or the widget should be the owner. As main does not use it afterwards, it should be probably owned by the widget.
So I'd suggest to either use an std::optional here (preferred) or use a unique_ptr.

Reply:
Here is how I got to this. When used in the real application, I would not need a pointer here at all, it could happily be a simple `RetrieveInstances` that is constructed in the constructor of `RetrieveInstancesWidget`. I only have a pointer type here, because I want to mock it in tests. So then the first thought is: `RetrieveInstances*` (1.), but then the actual object needs to live one hierarchy level higher (whoever owns `RetrieveInstancesWidget` also owns a `RetrieveInstances` and passes a pointer of it to the widget) (In the real case its actually `ConnectToStadiaWidget`). Which is okay i guess, but also not ideal, because `ConnectToStadiaWidget` would own this with the single purpose of passing it on to `RetrieveInstancesWidget`. The shared_ptr gives IMO a neat solution here, because in `ConnectToStadiaWidget`, I actually pass in a `std::unique_ptr` and therefore effectively transfer the ownership to `RetrieveInstancesWidget`. And in the tests I use a `shared_ptr` and can still access it outside the widget.

I guess another option would be to use `unique_ptr` instead and then do the following in the tests 
```
TestFixtureCtor(){
  std::unique_ptr<MockRetrieveInstances> mri = std::make_unique(...);
  test_ptr_to_mri_ = mri.get();
  RetrieveInstancesWidget widget{};
  widget.setRetrieveInstances(std::move(mri));
}
```
Then I could use the `test_ptr_to_mri_` for EXPECT_CALL and so on. This would work, but it seems a wrong thing to do, because the test fixture actively transfers the object into the widget, but also still holds a pointer to it. 

Maybe I am missing a reason why shared_ptr is not a good idea here, and then I would prefer to go with 1. (`RetrieveInstances*`). wdyt?

Reply:
Taked about offline and resolved.